lib/whitehall_importer/create_edition.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+class WhitehallImporter::CreateEdition
+  attr_reader :document, :whitehall_document, :whitehall_edition, :edition_number, :user_ids
+
+  def initialize(document, whitehall_document, whitehall_edition, edition_number, user_ids)
+    @document = document
+    @whitehall_document = whitehall_document
+    @whitehall_edition = whitehall_edition
+    @edition_number = edition_number
+    @user_ids = user_ids
+  end
+
+  def call
+    check_only_in_english
+
+    edition = if whitehall_edition["state"] == "withdrawn"
+                create_withdrawn_edition
+              else
+                create_edition
+              end
+
+    access_limit(edition)
+  end
+
+private
+
+  def check_only_in_english
+    raise WhitehallImporter::AbortImportError, "Edition has an unsupported locale" unless only_english_translation?
+  end
+
+  def only_english_translation?
+    whitehall_edition["translations"].count == 1 && whitehall_edition["translations"].last["locale"] == "en"
+  end
+
+  def english_translation
+    whitehall_edition["translations"].last
+  end
+
+  def create_withdrawn_edition
+    edition = create_edition("published")
+    set_withdrawn_status(edition)
+
+    edition
+  end
+
+  def create_edition(initial_state = whitehall_edition["state"])
+    create_event = create_history_event
```

Comment:
Maybe rename the `create_history_event` method, since it sounds like it's creating a history event. But again not necessary for initial merge.